### PR TITLE
Fix Summary result for Kubernetes application

### LINKED ISSUE #3007: Piped agent shows out a wrong Summary result for Kubernetes application in some cases
**What happened**:

**What you expected to happen**:

It results in a very long message like

```
Sync progressively because of updating image .....
```

**How to reproduce it**:

**Environment**:
- `piped` version:
- `control-plane` version:
- Others:

### PR DESCRIPTION
**Pull request by Hosshii**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #3007

**Does this PR introduce a user-facing change?**:

```release-note
Fix wrong summary result for Kubernetes application in case the image version was updated
```

### CODE REVIEW

**Comment by khanhtc1202** [comment]
Nice catch and well tested 👍 
nits, this PR provides a user-facing change, should write to the block something like: Fix wrong summary result for ... in case ...
/lgtm

File: pkg/app/piped/planner/kubernetes/kubernetes_test.go [hunk 1]
```diff
@@ -437,3 +437,60 @@ func TestDetermineVersion(t *testing.T) {
...
+				templateDiffs := diffNodes.FindByPrefix("spec.template")
+
+				msg, changed := checkImageChange(templateDiffs)
+
+				assert.Equal(t, tc.msg, msg)
+				assert.Equal(t, tc.changed, changed)
+			}
+		})
+	}
+
```
Comment (nghialv) [inline pkg/app/piped/planner/kubernetes/kubernetes_test.go:495]: nit: Remove empty line

**Comment by Hosshii** [comment]
Sorry, I fixed the wrong place, so I fixed it again.
And I changed user-facing change to `Fix wrong summary result for Kubernetes application in case the image version was updated`

**Comment by nghialv** [comment]
Sure.
/approve

**Pull request merged**